pyqtorch/modules/hamevo.py

```diff
@@ -165,47 +166,23 @@ def apply(self, state: torch.Tensor) -> torch.Tensor:
 
         return _apply_batch_gate(state, evol_operator, self.qubits, self.n_qubits, batch_size_h)
 
+class HamEvoType(Enum):
+    RK4 = HamEvo
+    EIG = HamEvoEig
+    EXP = HamEvoExp
 
 class HamiltonianEvolution(Module):
-    def __init__(self, qubits: Any, n_qubits: int, n_steps: int = 100):
+    def __init__(self, qubits: Any, n_qubits: int, n_steps: int = 100, type: HamEvoType = HamEvoType.RK4):
         super().__init__()
         self.qubits = qubits
         self.n_qubits = n_qubits
         self.n_steps = n_steps
+        if isinstance(type, str):
+            type = HamEvoType[type.upper()]
```

Comment:
You could add a `try...except` block here to catch the case where the user passes an invalid string and send a clear error message.